File: Front/.Net Maui/BetFriends.Domain.UnitTests/Implems/MockRegisterPresenter.cs
using BetFriends.Domain.Features.Register;

namespace BetFriends.Domain.UnitTests.Implems;

internal class MockRegisterPresenter : IRegisterOutputPort
{
    public string Error { get; private set; }
    public string Id { get; private set; }

    public void FieldEmpty()
    {
        Error = "field empty";
    }

    public void PasswordsNotEqual()
    {
        Error = "passwords are not equal";
    }

    public void Present(string id)
    {
        Id = id;
    }
}

File: Front/TS/tests/unit/implems/MockRegisterPresenter.ts
import { IRegisterOutputPort } from "../../../src/domain/features/RegisterHandler";

export class MockRegisterPresenter implements IRegisterOutputPort {
    PasswordsAreDifferent(): void {
        this.Error = "passwords are different"
    }
    FieldIsEmpty(): void {
        this.Error = "field is empty"
    }
    UserRegistered(userId: string): void {
        this.UserId = userId;
    }
    UserId?: string;
    Error?: string

}